include/picongpu/particles/pusher/particlePusherAxel.hpp

```diff
@@ -82,8 +84,17 @@ namespace picongpu
                 using MomType = momentum::type;
                 MomType mom = particle[momentum_];
 
-                auto bField = functorBField(pos);
-                auto eField = functorEField(pos);
+                const auto bField = functorBField(pos);
+                const auto eField = functorEField(pos);
+
+                // update probe field if particle contains required attributes
+                pmacc::meta::executeIf<pmacc::traits::HasIdentifier<T_Particle, probeB>::type::value>(
+                    [&bField](auto&& par) { par[probeB_] = bField; },
+                    particle);
+
+                pmacc::meta::executeIf<pmacc::traits::HasIdentifier<T_Particle, probeE>::type::value>(
+                    [&eField](auto&& par) { par[probeE_] = eField; },
+                    particle);
```

Comment:
Also discussed offline a few days ago. This piece is probably better as a functor/function called from all pushers.

Reply:
This will be part of a follow-up PR to avoid breaking the already well-tested implementation.